x-pack/platform/plugins/shared/security/server/authentication/providers/oidc.ts

```diff
@@ -304,13 +304,21 @@ export class OIDCAuthenticationProvider extends BaseAuthenticationProvider {
       // user usually doesn't have `cluster:admin/xpack/security/oidc/prepare`.
       // We can replace generic `transport.request` with a dedicated API method call once
       // https://github.com/elastic/elasticsearch/issues/67189 is resolved.
-      const { state, nonce, redirect } =
+      const { state, nonce, realm, redirect } =
         (await this.options.client.asInternalUser.transport.request({
           method: 'POST',
           path: '/_security/oidc/prepare',
           body: params,
         })) as any;
 
+      if (realm && this.realm && realm !== this.realm) {
+        this.logger.debug(
```

Comment:
__optional nit:__ our logger is already configured with provider type and name, so these are already included in the logs, so maybe something like this to not duplicate info?

> Provider is configured with the "${this.realm}" realm and isn't compatible with the "${realm}" realm used by Elasticsearch to prepare the authentication request. Skipping provider…